.github/workflows/github-actions-workflow.yaml

```diff
@@ -0,0 +1,83 @@
+name: drone-gae
+
+on:
+  push:
+    branches: [ "main" ]
+    tags:
+      - '*'
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  GO111MODULE: on
+  CGO_ENABLED: 0
+ 
+jobs:
+   build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+
+    - name: Test
+      run: |
+        go mod download
+        go test -cover -vet all
+    
+    - name: Build
+      run: |
+        go build -a -ldflags "-X main.version=${{ inputs.github_ref_name }} -X main.rev=${{ inputs.github_sha }}"
+        
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: nytimes/drone-gae
+        tags: |
+          type=raw,value=latest,enable={{is_default_branch}}
+          type=ref,event=tag
+  
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    
+    - name: Build and push dry run
+      if: ${{ github.event_name == 'pull_request' }}
+      uses: docker/build-push-action@v4
+      with:
+        context: .
+        platforms: linux/amd64
+        push: false
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+    
+    - run: echo ${GITHUB_REF}
+    
+    - name: Build and push
+      if: startsWith(github.ref, 'refs/tags/v')
+      uses: docker/build-push-action@v4
+      with:
+        context: .
+        platforms: linux/amd64
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
```

Comment:
This doesn't need to be two steps - we can use a single step with `${{ github.event_name == 'pull_request' }}` on the `push` parameter. It will only push when there are commits to `main` and the `tag` action.

The Drone build pushes for both main and tags (main pushing `latest` and tag pushing whatever the tagged version is).

```
    - name: Build and push
      uses: docker/build-push-action@v4
      with:
        context: .
        platforms: linux/amd64
        push: ${{ github.event_name != 'pull_request' }}
        tags: ${{ steps.meta.outputs.tags }}
        labels: ${{ steps.meta.outputs.labels }}
```

Reply:
Ok, combing into one step.